src/main/marketplace/tax-treaties/tax-treaties-service.js

```diff
@@ -58,26 +62,40 @@ export class TaxTreatiesService {
 	async fetchTaxTreatiesFlagtheory() {
 		try {
 			let fetched = await request.get({ url: FLAGTHEORY_TAX_TREATIES_ENDPOINT, json: true });
-			return fetched[0].map(entity =>
-				_.mapKeys(_.omit(entity, 'id'), (value, key) => _.camelCase(key))
-			);
+			let data = [];
+			for (const entity of fetched[0]) {
+				const item = await mapKeysAsync(_.omit(entity, 'id'), (value, key) =>
+					_.camelCase(key)
+				);
+				data.push(item);
+				await setImmediatePromise();
+			}
+			return data;
 		} catch (error) {
 			log.error(`fetchTaxTreatiesFlagtheory: ${error}`);
 			return [];
 		}
 	}
 	start() {
-		this.schedulerService.queueJob(null, TAX_TREATIES_SYNC_JOB);
+		const now = new Date();
+		this.schedulerService.queueJob(
+			null,
+			TAX_TREATIES_SYNC_JOB,
+			now.setMinutes(now.getMinutes() + 1)
+		);
 	}
-
 	loadTaxTreaties() {
 		return TaxTreaties.findAll();
 	}
-	upsert(upsert) {
-		return TaxTreaties.bulkUpsert(upsert);
+	async upsert(upsert) {
+		const chunks = arrayChunks(upsert, 100);
+		for (const c of chunks) {
+			TaxTreaties.bulkUpsert(c);
```

Comment:
should be `await` here, the promise should not be ignored

Reply:
Also, maybe all of them need to be wrapped in a transaction? So that if upsert of one chunk fails, all of the operation will be failed ?

Reply:
Another thing, it seems to me like a model level operation, instead of service level. It is atomic to `bulkUpsert` operation in my opinion

Reply:
Agree 💯 , the service should be agnostic to the upsert model implementation details (and a yes a transaction was clearly missing here).